packages/wonder-blocks-data/util/response-cache.test.js

```diff
@@ -111,117 +112,597 @@ describe("./response-cache.js", () => {
+            });
+        });
+    });
+
+    describe("#remove", () => {
+        it("should return false if nothing was removed", () => {
             // Arrange
-            const cache = new ResponseCache();
+            const cache = new ResponseCache({
+                MY_HANDLER: {
+                    MY_OTHER_KEY: {data: "data!"},
+                },
+            });
+            const customCache = {
+                store: jest.fn(),
+                retrieve: jest.fn(),
+                remove: jest.fn(),
+                removeAll: jest.fn(),
+            };
             const fakeHandler: IRequestHandler<string, string> = {
                 getKey: () => "MY_KEY",
                 type: "MY_HANDLER",
                 shouldRefreshCache: () => false,
                 fulfillRequest: jest.fn(),
-                cache: null,
+                cache: customCache,
             };
 
             // Act
-            cache.cacheData(fakeHandler, "options", "data");
-            const result = cache.getEntry(fakeHandler, "options");
+            const result = cache.remove(fakeHandler, "optionsA");
 
             // Assert
-            expect(result).toStrictEqual({data: "data"});
+            expect(result).toBeFalsy();
         });
 
-        it("should replace the entry in the handler subcache", () => {
+        it("should return true if something was removed from framework cache", () => {
             // Arrange
             const cache = new ResponseCache({
                 MY_HANDLER: {
-                    MY_KEY: {error: "Oh no!"},
+                    MY_KEY: {data: "data!"},
                 },
             });
+            const customCache = {
+                store: jest.fn(),
+                retrieve: jest.fn(),
+                remove: jest.fn(),
+                removeAll: jest.fn(),
+            };
             const fakeHandler: IRequestHandler<string, string> = {
                 getKey: () => "MY_KEY",
                 type: "MY_HANDLER",
                 shouldRefreshCache: () => false,
                 fulfillRequest: jest.fn(),
-                cache: null,
+                cache: customCache,
             };
 
             // Act
-            cache.cacheData(fakeHandler, "options", "other_data");
-            const result = cache.getEntry(fakeHandler, "options");
+            const result = cache.remove(fakeHandler, "optionsA");
 
             // Assert
-            expect(result).toStrictEqual({data: "other_data"});
+            expect(result).toBeTruthy();
         });
-    });
 
-    describe("#cacheError", () => {
-        it("should add the entry to the cache", () => {
+        it("should return true if something was removed from custom cache", () => {
             // Arrange
-            const cache = new ResponseCache();
+            const cache = new ResponseCache({
+                MY_HANDLER: {
+                    MY_OTHER_KEY: {data: "data!"},
```

Comment:
The reason why this isn't `MY_KEY` is because there's no way to get `MY_KEY` into a custom cache without it first being removed from the framework cache?

Reply:
I don't understand the question.

Reply:
In `getEntry` we do `delete handlerCache[key];` if there's a custom cache.  As a result we'd never be in a state where we'd have:
```
MY_HANDLER: {
  MY_KEY: {data: "data!"},
}
```
in both the framework cache and the custom cache since the only way for something to get into the custom cache is by calling `getEntry()` which transfers the entry from the framework cache to the custom cache.  At least that's my understanding of things.  I wanted to confirm that that's actually the case and that the data in the test cases reflects that invariant.

Reply:
There are two ways things go into custom cache.

1. When hydrating on the client from SSR result, such that the framework cache has something in it that should be in the custom cache (done on `getEntry`)
2. When a client-side request is handled and the result is cached

Framework cache is only used client-side when there is no custom cache, or during SSR.

As for this test case, all that data is a red herring.

The custom cache implementation returns `true` and so our framework should be doing so. The data is just present for a complete test case. I'll see if I can make that clearer.